.github/workflows/docsbuild-release.yaml

```
@@ -0,0 +1,85 @@
+name: Create an Ansible Release Porting Guide PR
+on:
+  workflow_dispatch:
+    inputs:
+      ansible-version:
+        description: >-
+          Release Version. Example: 11.1.0
+        required: true
+env:
+  CI_COMMIT_MESSAGE: >-
+    Add Ansible community "${{ inputs.ansible-version }}" porting guide
+
+jobs:
+  build:
+    name: >-
+      Build Porting-Guide (${{ inputs.ansible-version }})
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Retrieve the major version component
+        id: compute-version
+        run: |
+          echo -n 'major-version=' >> "${GITHUB_OUTPUT}"
+          echo '${{ inputs.ansible_version }}' | sed 's#\([0-9]\+\)\..*#\1#' >> "${GITHUB_OUTPUT}"
+
+      - name: Check out ansible-documentation
+        uses: actions/checkout@v3
+        with:
+          path: ansible-documentation
+          ref: devel
+
+      - name: Checking out to a new branch & setting the user details
+        working-directory: ansible-documentation
+        run: |
+          git checkout -b "publish-${{ inputs.ansible-version }}"
+          git config --global user.name "Github Actions"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+# Get the Porting guide from the ansible-build-data git repo
+
+      - name: Check out ansible-build-data
+        uses: actions/checkout@v3
+        with:
+          repository: ansible-community/ansible-build-data
+          ref: main
```

Also, I see that the repo has tags so it appears, it's possible to grab the properly tagged version exactly and retrieve just what we're interested in:
```suggestion
        uses: actions/checkout@v4
        with:
          repository: ansible-community/ansible-build-data
          ref: ${{ inputs.ansible-version }}
          sparse-checkout: ${{ steps.compute-version.outputs.major-version }}/porting_guide_${{ steps.compute-version.outputs.major-version }}.rst
```

I also added a filter to only get that document that we're interested in and updated the action to v4 which is what seems to have introduced support for sparse checkouts and doesn't use an outdated NodeJS version.